### ISSUE TITLE
We should test against Python 3.7

### ISSUE THREAD

**Issue by Zac-HD** [HypothesisWorks/hypothesis#1148]
See https://www.python.org/downloads/release/python-370b2/

As well as helping to test Python itself, we get to check that nothing will suddenly break come June.

3.7-related considerations for Hypothesis:

- The delayed evaluation of type hints should be fine (`typing.get_type_hints` still does the heavy lifting there), but there may be some other changes to the typing module that require tweaks to code or tests around strategy inference.  #1004 still looks hopeless.
- `builds()` should introspect the new dataclasses.  See #954 for `attrs` equivalent.

**Comment by Zac-HD** [comment]
Welp.  Too late for pre-releases, but we can use the issue to track general progress on 3.7 support 🚀 🙄

**Comment by DRMacIver** [comment]
See https://github.com/HypothesisWorks/hypothesis/pull/1369

Basically: This isn't going to happen any time soon.

The following scenarios would allow it to happen:

1. Travis makes it possible to test on a modern version of Linux
2. CPython makes it easy to statically link OpenSSL
3. Someone else (pyenv?) does the wrangling that would allow us install this with a custom OpenSSL easily.
4. We switch our build environment over to using anaconda somehow.

Basically I'm grumpy and pissed off about this whole thing and super not invested in supporting Python 3.7, so I'm declaring this to be somebody else's problem.

**Comment by Zac-HD** [comment]
We now have 3.7 running on Travis, thanks to #1376 (:heart_eyes: @asottile)

It currently has an allowed failure because of [a console unicode issue here](https://github.com/HypothesisWorks/hypothesis/blob/b1518f16f9b60969dc0419ea30ff25f1df6ec674/hypothesis-python/tests/pytest/test_capture.py#L75-L84), thanks to [PEP 538](https://www.python.org/dev/peps/pep-0538/).  All we need to do for *this* is to accept the character (`ϩ`) even if it's not displayed as `\u03e9`!

I do expect some other problems to come up though :smile:
